packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/edge/CustomEdge.types.ts

```diff
@@ -0,0 +1,20 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo and others.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import { GQLLabel } from '../../graphql/subscription/labelFragment.types';
+
+export interface CustomEdgeData {
+  beginLabel: GQLLabel;
+  centerLabel: GQLLabel;
+  endLabel: GQLLabel;
```

Comment:
Use an internal type like `Label` instead of just the GQLLabel. There are things to process before such as the style. See how convertDiagram is in charge of converting the raw GQL style data to CSSProperties for example.